Sources/StreamChat/ChatClient.swift

```diff
@@ -637,8 +555,8 @@ extension ChatClient: ConnectionStateDelegate {
             connectionId = id
         case let .disconnected(source):
             if let error = source.serverError,
-               error.isTokenExpiredError {
-                refreshToken(error: error, completion: nil)
+               error.isInvalidTokenError {
+                refreshToken(completion: nil)
```

Comment:
That is not nice 😕 

The thing is we don't have token refreshing logic encapsulated so there's no easy way to move it from `ChatClient` to `ConnectionRecoveryHandler`. That's why in case of expired token `ConnectionRecoveryHandler ` does nothing (since `isAutomaticReconnectionEnabled` returns false) and just waits for the chat client to refetch the token and initiate reconnection by itself.